datahub-web-react/src/app/permissions/policy/ManagePolicies.tsx

```diff
@@ -258,17 +262,28 @@ export const ManagePolicies = () => {
             title: `Delete ${policy?.name}`,
             content: `Are you sure you want to remove policy?`,
             onOk() {
-                deletePolicy({ variables: { urn: policy?.urn as string } }); // There must be a focus policy urn.
-                analytics.event({
-                    type: EventType.DeleteEntityEvent,
-                    entityUrn: policy?.urn,
-                    entityType: EntityType.DatahubPolicy,
-                });
-                message.success('Successfully removed policy.');
-                setTimeout(() => {
-                    policiesRefetch();
-                }, 3000);
-                onCancelViewPolicy();
+                removeFromListPoliciesCache(client, policy?.urn , page, pageSize, query);
+                deletePolicy({ variables: { urn: policy?.urn as string } }) // There must be a focus policy urn.
+                .then(() => {
+                    analytics.event({
+                        type: EventType.DeleteEntityEvent,
+                        entityUrn: policy?.urn,
+                        entityType: EntityType.DatahubPolicy,
+                    });
+                    message.success('Successfully removed policy.');
+                    const newRemovedUrns = [...removedUrns, policy?.urn];
+                    setRemovedUrns(newRemovedUrns);
+                    setTimeout(() => {
+                        policiesRefetch?.();
+                    }, 3000);
+                    onCancelViewPolicy();
+                })
+                .catch((e: unknown) => {
+                    message.destroy();
+                    if (e instanceof Error) {
+                        message.error({ content: `Failed to remove Policy: \n ${e.message || ''}`, duration: 3 });
```

Comment:
Let's just say: Failed to remove policy! An unexpected error occurred.

Reply:
Done